### DISCUSSION TITLE
This is just great!! I just have 1 question

### DISCUSSION THREAD

**Discussion by Voxel3DPrinting** [digitalninja-ro/klipper-neopixel#18] [Q&A]
Hi,

Absolutely love this macro, welldone @digitalninja-ro!!

Just to be absolutely sure;
If I want to enable the progressbar during printing, I need to add;
"NEOPIXEL_DISPLAY LED="case_leds" TYPE=print_percent MODE=progress"
 to my start gcode, just before "STATUS_PRINTING"
 
 Also,
 I want the ledstrip to be white as standard, changing to green depending on progress.
 
 Can I achieve that by setting the code to;
 [display_template led_print_percent_progress]
param_led_num:  0
param_led_total: 1
text:
    {% set ratio = printer.virtual_sdcard.progress %}
    {% set led_ratio   = param_led_num|float / param_led_total %}
    {% if ratio > led_ratio %}
        0.0, {led_ratio}, 0.0
    {% else %}
        1.0, 1.0, 1.0
    {% endif %}

**Comment by digitalninja-ro** [comment]
Hi. Thanks for your feedback.
1. I recommend to add the neopixel macro call in your START_PRINT macro. It this way will work with any slicer
2. Yes you can use your code to achieve what you want. But I'll use 0.1, 0.1, 0.1 to decrease the light in inactive state.

**Reply by Voxel3DPrinting** [reply]
Great! that did work!

However, I do have some other issues now;

**Question 1;**
- When done printing, all LEDs are (obviously) green, which is correct. However, If I then use another macro/command to try to turn them white for the camera, set them to standby-mode, or turn them off, it looks like it is locked in the fully green state and does not respond to any input... Is this fixable??
macro's; 
[gcode_macro CASE_LED_on]
gcode:
    SET_LED LED=case_leds RED=1 GREEN=1 BLUE=1

[gcode_macro CASE_LED_off]
gcode:
    SET_LED LED=case_leds RED=0 GREEN=0 BLUE=0

[gcode_macro CASE_LED_standby]
gcode:
    SET_LED LED=case_leds RED=0.2 GREEN=0.2 BLUE=0.2
    
**Question 2;**
 I often monitor my prints abroad, for which I use my camera. I use the same led strip for the progress as a light to brighten up the scene. Would it be possible to temporarily switch this strip to full white during printing/ progress?
 Like a macro that is only possible during printing, which switches the light to full white for 1minute, after which it turns back to the progress?

**Reply by digitalninja-ro** [reply]
1. As long you have assigned a template to a led, you can't change the led, the template does.  Use the [beta](https://github.com/digitalninja-ro/klipper-neopixel/tree/beta) version where I implemented a macro that [removes](https://github.com/digitalninja-ro/klipper-neopixel/tree/beta#clear-template) the template and turn leds off.
2. You can write a simple macro in which you remove the template (see answer 1), turn the led on with SET_LED commend for 1 min and then load back the template for progress print (you cand use [delayed_gcode](https://www.klipper3d.org/G-Codes.html#delayed_gcode)). When you connect from abroad just execute the macro and look at your print.

**Reply by Voxel3DPrinting** [reply]
I can Turn on the led now, and enable it via a macro during printing. 

However, if I send "NEOPIXEL_DISPLAY LED=case_leds TYPE=print_percent MODE=[progress]" again after that, it does not return to the progress bar.

Anything I can try?

**Reply by digitalninja-ro** [reply]
The syntax is:
```NEOPIXEL_DISPLAY LED=case_leds TYPE=print_percent MODE=progress```
